### DISCUSSION TITLE
FEA: Support for Mailer Previews

### DISCUSSION THREAD

**Discussion by drush** [lookbook-hq/lookbook#733] [Feature requests/ideas]
Lookbook is a better, modern preview implementation for Rails applications. While it handles Pages and Components out-of-the-box, it does not currently include support for Mailers. Mailers and Mailer Previews share almost identical semantics to Components for developers in view/develop/test workflows.

This feature request is to add support for Mailer Previews as a first-order feature of Lookbook.

Developers have many options for Mailer Previews including built-in `/rails/mailers/` preview route, Maily, letter_opener, but removing these dependencies for unified previews through Lookbook is beneficial from a code complexity and security standpoint.  The UI for Lookbook also offers a modern, rapid workflow.

PROPOSED
- Lookbook to detect classes inherited from `ActionMailer::Preview` automatically to add to the component menu
- Use a view wrapper for mailer previews to add space for meta headers (to, from, subject, etc) and allow for the mailer to be self-contained HTML (rather than using a shared layout)

REFERENCE
- @baygeldin Has a nice [POC for Mailer Previews](https://evilmartians.com/chronicles/viewcomponent-in-the-wild-supercharging-your-components#preview-for-mailers) in this post

**Comment by allmarkedup** [comment]
Hey @drush, I'm so sorry for the ridiculously slow reply.

I have to say that I love this idea. It certainly feels like a good fit for Lookbook and would be a really nice feature.

I have personally never worked with mailer previews but I believe some of the original design of the native ViewComponent preview system was inspired by the way that they work. And the article you've linked to shows it's definitely possible so I don't think supporting these kind of previews would be too much of a stretch.

Unfortunately I have very little spare time at the moment to implement new features but I will keep this in mind for when I next get a chance to do some more work on Lookbook. If I get a first pass together I'll let you know so you can kick the tyres on it a bit :)

Thanks again for the suggestion and apologies again for being so slow to get back to you.
